velero-plugins/common/shared.go

```diff
@@ -87,18 +84,39 @@ func getMetadataAndAnnotations(item runtime.Unstructured) (metav1.Object, map[st
 	return metadata, annotations, nil
 }
 
-func GetServerVersion() (*version.Info, error) {
+// returns major, minor versions for kube
+func GetServerVersion() (int, int, error) {
 	client, err := clients.DiscoveryClient()
 	if err != nil {
-		return nil, err
+		return 0, 0, err
 	}
 	version, err := client.ServerVersion()
 	if err != nil {
-		return nil, err
+		return 0, 0, err
 	}
-	if strings.HasSuffix(version.Minor, "+") {
-		version.Minor = strings.TrimSuffix(version.Minor, "+")
+
+	// Attempt parsing version.Major/Minor first, fall back to parsing gitVersion
+	major, err1 := strconv.Atoi(version.Major)
+	minor, err2 := strconv.Atoi(strings.Trim(version.Minor, "+"))
+
+	if err1 != nil || err2 != nil {
+		// gitVersion format ("v1.11.0+d4cacc0")
+		r, _ := regexp.Compile(`v[0-9]+\.[0-9]+`)
+		valid := r.MatchString(version.GitVersion)
+		if !valid {
+			return 0, 0, errors.New("gitVersion does not match expected format")
+		}
+		majorMinorArr := strings.Split(strings.Split(version.GitVersion, "v")[1], ".")
```

Comment:
Assume this is safe since `version.GitVersion` is checked prior to this with the valid check.